Sources/PromotedCore/ClientConfig/_ObjCClientConfig.swift

```diff
@@ -128,6 +128,19 @@ public final class _ObjCClientConfig: NSObject {
     didSet { validateEnum(&metricsLoggingWireFormat, defaultValue: .binary) }
   }
 
+  /// How to format the JSON message. You can use this to put the JSON
+  /// object for the event inside another JSON object. Use `${event}`
+  /// as a placeholder for the JSON event. For example, a value of the
+  /// following string:
+  /// ```swift
+  /// { "foo": "bar", "event": ${event} }
+  /// ```
+  /// Will log the JSON string:
+  /// ```json
+  /// { "foo": "bar", "event": { "action": "NAVIGATE", "content_id": ... } }
+  /// ```
+  @objc public var metricsLoggingJSONFormatString: String = "${event}"
```

Comment:
Interesting.  Not for this PR - I don't remember the design of this code.  If we have to have to get very custom outputs, could we also make the serde and rpc logic customizable?  E.g. make it a sink that can be swapped in.

I'm fine with this approach right now since most people will want a small set of configurations.

Reply:
I agree. We already support custom serve and RPC implementations, but they require more code from the client. This is kind of a shortcut to allow the JSON to be specified as the child in another JSON object.

If there was much more customization than this required, I would say the client should write their own `NetworkConnection` interface instead.